BATCH-2515: Integrate SAP Hana as supported Spring Batch database

breglerj opened the pull request:
This commit adds SAP HANA as a supported Spring Batch database, enabling
developers to seamlessly move their existing Spring Batch projects to SAP HANA
or easily starting new Spring Batch projects on SAP HANA.

This commit contains the following changes:

- Add SAP HANA to the DatabaseType enum
- Add HanaPagingQueryProvider and tests
- Add properties files for SAP HANA
- Increase Spring version to 5.0.5 which is the first version to include
  HanaSequenceMaxValueIncrementer (see [https://jira.spring.io/browse/SPR-16558](https://jira.spring.io/browse/SPR-16558))

benas commented on spring-batch-core/src/main/resources/org/springframework/batch/core/migration/migration-hana.sql:
```diff
@@ -0,0 +1,28 @@
+
+-- insert script that 'copies' existing batch_job_params to batch_job_execution_params
+-- sets new params to identifying ones
+-- verified on h2, 
```
> verified on h2

Is there a way to verify DDL scripts on a real SAP HANA instance? We have introduced Docker-based integration tests to verify DDL scripts on non-embeddable databases (see [MySQLJobRepositoryIntegrationTests](https://github.com/spring-projects/spring-batch/blob/main/spring-batch-core/src/test/java/org/springframework/batch/core/test/repository/MySQLJobRepositoryIntegrationTests.java) as an example). I see there is a Docker image for SAP HANA express edition (which requires an DockerHub account, but that's not an issue). Could you please add a Docker-based test for HANA?

breglerj replied:
I have added an integration test based on the SQL Server test. Since the [PR ](https://github.com/testcontainers/testcontainers-java/pull/3017)for adding HANA to testcontainers is still open I've added a simple version of the HANA container class to the test project.

benas commented on spring-batch-core/src/main/resources/org/springframework/batch/core/migration/migration-hana.sql:
```diff
@@ -0,0 +1,28 @@
+
```
I'm not sure we need this "migration" script, as there was no support for SAP HANA previously. Migration scripts in the `org/springframework/batch/core/migration` package are typically for the modifications to the existing DDL scripts of an already supported database and which are required to migrate to a newer version of Spring Batch.

breglerj replied:
I have removed the migration script.

benas commented:
Hi @breglerj ,

Thank you for your PR! We are planning to add support for SAP HANA as a job repository in the next version, v5. The changes look good to me 👍  My only concern here is with testing. I'm not familiar with HANA so I would appreciate your help in making sure we can test this support on a real HANA instance. Please take a look at my comments above. Thank you.

breglerj commented:
> Hi @breglerj ,
> 
> Thank you for your PR! We are planning to add support for SAP HANA as a job repository in the next version, v5. The changes look good to me 👍 My only concern here is with testing. I'm not familiar with HANA so I would appreciate your help in making sure we can test this support on a real HANA instance. Please take a look at my comments above. Thank you.

Hi @benas ,

I've updated this PR and incorporated your comments. Please let me know if you need anything else.

Thanks.

benas commented:
Rebased and merged as 9388abb58983c32dae0e795d3173de951d8b32bc. I introduced some minor updates in 968a5b5b2e12a747e140da50d8cac007a4bdecb7.

Thank you for this great addition!
